app/src/main/java/com/vasilevkin/catsanddogs/features/animalList/view/ui/MainActivity.kt

```diff
@@ -0,0 +1,99 @@
+package com.vasilevkin.catsanddogs.features.animalList.view.ui
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.vasilevkin.catsanddogs.R
+import com.vasilevkin.catsanddogs.delegateadapter.diff.DiffUtilCompositeAdapter
+import com.vasilevkin.catsanddogs.delegateadapter.diff.IComparableItem
+import com.vasilevkin.catsanddogs.features.animalList.IMainContract
+import com.vasilevkin.catsanddogs.features.animalList.view.adapter.BigViewpagerDelegateAdapter
+import com.vasilevkin.catsanddogs.features.animalList.view.adapter.LongHorizontalDelegateAdapter
+import com.vasilevkin.catsanddogs.features.animalList.view.adapter.SquareDelegateAdapter
+import com.vasilevkin.catsanddogs.models.localModels.SquareCatLocalModel
+import kotlinx.android.synthetic.main.activity_main.*
+import org.koin.android.ext.android.inject
+import org.koin.core.parameter.parametersOf
+
+
+class MainActivity : AppCompatActivity(), IMainContract.View {
+
+    private var cats: List<IComparableItem> = emptyList()
+    private val presenter: IMainContract.Presenter by inject { parametersOf(this) }
+
+    private val diffAdapter by lazy {
```

Comment:
by lazy is heavy operation - try to use lateinit or nullable fields

Reply:
fixed.